pkg/sql/sessiondata/session_data.go

```diff
@@ -76,6 +77,8 @@ type SessionData struct {
 	SequenceState *SequenceState
 	// DataConversion gives access to the data conversion configuration.
 	DataConversion DataConversionConfig
+	// DurationAdditionMode enables math compatibility options to be enabled.
+	DurationAdditionMode duration.AdditionMode
```

Comment:
Put a TODO here to remove this in the 2.2 release cycle, since at that point we know it's safe to remove the old behavior.